src/GitHub.Exports/Services/VSServices.cs

```diff
@@ -68,6 +73,50 @@ public void ActivityLogWarning(string message)
             }
         }
 
+        /// <summary>Open a repository in Team Explorer</summary>
+        /// <remarks>
+        /// There doesn't appear to be a command that directly opens a target repo.
+        /// Our workaround is to create, open and delete a solution in the repo directory.
+        /// This triggers an event that causes the target repo to open. ;)
+        /// </remarks>
+        /// <param name="repoPath">The path to the repository to open</param>
+        /// <returns>True if a transient solution was successfully created in target directory (which should trigger opening of repository).</returns>
+        public bool TryOpenRepository(string repoPath)
+        {
+            bool solutionCreated = false;
+
+            try
+            {
+                var dte = (DTE)serviceProvider.GetService(typeof(DTE));
+                var os = (IOperatingSystem)serviceProvider.GetService(typeof(IOperatingSystem));
+
+                try
+                {
+                    dte.Solution.Create(repoPath, TempSolutionName);
+                    solutionCreated = true;
+
+                    // Don't create a .sln file when we close.
+                    dte.Solution.Close(false);
+                }
+                finally
+                {
+                    // Clean up the dummy solution's subdirectory inside `.vs`.
+                    var vsDirPath = Path.Combine(repoPath, ".vs", TempSolutionName);
+                    var vsDir = os.Directory.GetDirectory(vsDirPath);
+                    if (vsDir.Exists)
+                    {
+                        vsDir.Delete(true);
```

Comment:
Don't want to delete `.vs` itself, we don't own it and VS uses it for its own things. If we're not creating an actual file, then there's no need to delete anything, is there?